src/sentry/utils/session_store.py

```diff
@@ -71,7 +80,11 @@ def clear(self):
             return
 
         self._client.delete(self.redis_key)
-        del self.request.session[self.session_key]
+
+        session = self.request.session
+        del session[self.session_key]
+        if hasattr(session, "modified"):
```

Comment:
what is 
```
        if hasattr(session, "modified"):
            session.modified = True
```
doing?

Reply:
Replicating https://github.com/getsentry/sentry/pull/26883/files#diff-e0494c56201878812ba2d58e03a3e6310b0da0b3ed3f5c8ee6c0c9fa243d54aaL89. To be honest, beyond that I don't know.

When I put this here, I assumed it was harmless, but the `hasattr` was because it turned out to break some other cases. Thanks for provoking a second look -- I'm going to add a new hook for this.